### ISSUE TITLE
`from` is now a reserved word in for loop declarations

### ISSUE THREAD

**Issue by IvanVergiliev** [jashkenas/coffeescript#4390]
Apparently the addition of the `for...from` loop breaks code with variables called `from`. Obviously this can't be fixed, but it should at least be documented as a breaking change in the Changelog.

Also, I had a hard time finding a set of all reserved words in CoffeeScript - this would also be a useful addition, IMO.

**Comment by GeoffreyBooth** [comment]
`from` is only a keyword in the context of an import/export statement or for loop. I can use it as a variable: http://coffeescript.org/#try:from%20%3D%202

Can you post example code that worked in 1.11.1 but broke in 1.12.0?

**Comment by IvanVergiliev** [comment]
```coffeescript
lines = [
  {from: 0, to: 10}
  {from: 3, to: 17}
]

for {from, to} in lines
  console.log from
```

I also noticed that I can use it as a variable name sometimes. Apparently this case breaks because it's part of a `for` loop.

**Comment by GeoffreyBooth** [comment]
@atg, do you feel like looking into this?

**Comment by vendethiel** [comment]
Oh, it's just a really old thing the compiler's lexer does. That's just `@seenFor` parsing stuff differently between a `for` and a `from/of/in`...

**Comment by distracteddev** [comment]
@IvanVergiliev Quick question, if adding "from" support is clearly a breaking change, why did the project decide to publish it without a major version bump? Should I expect this project to respect semver in the future?

**Comment by vendethiel** [comment]
@distracteddev where did it say it was "clearly a breaking change"? @GeoffreyBooth even offered to look at 1.12 regressions.

(Also, https://gist.github.com/jashkenas/cbd2b088e20279ae2c8e)

**Comment by distracteddev** [comment]
My bad, assumed adding "from" support was known to be a breaking change. I see now that this wasn't intended to be the case.

Thanks for the note on semver.

**Comment by GeoffreyBooth** [comment]
Fixed in #4393. We will be releasing 1.12.1 soon.
